contracts/DCAStrategies/DCAStrategies/DCAStrategiesPositionsHandler.sol

```diff
@@ -241,19 +247,17 @@ abstract contract DCAStrategiesPositionsHandler is IDCAStrategiesPositionsHandle
           address(this),
           new IDCAPermissionManager.PermissionSet[](0)
         );
-        // TODO: search for a way of changing `_task.positionId` for `positionId`
-        if (i < _position.positions.length) {
-          _userPositions[_task.positionId].positions[i] = _newPositionId;
-        } else {
-          _userPositions[_task.positionId].positions.push(_newPositionId);
-        }
       }
 
+      _newPositions[i] = _task.positionId;
+
       unchecked {
         i++;
       }
     }
 
+    _userPositions[_positionId].positions = _newPositions;
```

Comment:
I noticed some strange behaviors that caused an ID to be overwritten (i.e. a position that was reused was overridden by a new position just deposited). Also, it made the code much more simpler 😅

Reply:
I agree that the code is **much much** simpler, but I think it will end up being more expensive. Because in the case of increase/reduce/nothing, you are re-writing the same value that was already there

Reply:
What do you suggest?